project-clone/internal/git_operations.go

```diff
@@ -88,11 +100,20 @@ func SetupRemotes(repo *git.Repository, project *dw.Project) error {
 }
 
 // CheckoutReference sets the current HEAD in repo to point at the revision and remote referenced by checkoutFrom
-func CheckoutReference(repo *git.Repository, checkoutFrom *dw.CheckoutFrom) error {
+func CheckoutReference(repo *git.Repository, project *dw.Project) error {
+	checkoutFrom := project.Git.CheckoutFrom
 	if checkoutFrom == nil {
 		return nil
 	}
-	remote, err := repo.Remote(checkoutFrom.Remote)
+	var defaultRemoteName string
+	if len(checkoutFrom.Remote) == 0 && len(project.Git.Remotes) == 1 {
```

Comment:
Slightly more Golang-style:
```suggestion
	if checkoutFrom.Remote == "" && len(project.Git.Remotes) == 1 {
```

Does this catch the case `checkoutFrom.Remote == "" && len(project.Git.Remotes) > 1` or is that handled earlier?

Reply:
this case is handled before

```golang
if needClone {
...
   repo, err := internal.CloneProject(&project).  // <--here we do the check
...
  if err := internal.CheckoutReference(repo, &project); err != nil {
```